core/src/avm1/activation.rs

```diff
@@ -544,18 +546,30 @@ impl<'a, 'gc> Activation<'a, 'gc> {
         }
     }
 
+    fn stack_push(&mut self, mut value: Value<'gc>) {
+        if let Value::Object(Object::StageObject(s)) = value {
```

Comment:
Do we need the ability to push to the AVM1 stack (without* this logic running? If not, could we move this logic into the `push` method, and have it take an `Activation`?

Reply:
We don't really need to be able to push without this running, so moving it would be possible, but because we need a mutable borrow of activation inside of `push` to create a `MovieClipReference` as well as a mutable borrow of `self.context.avm1` to call push it would likely be a much more invasive change

Reply:
Just adding that the only solution I can think of, would be either removing `avm1.push` and switching everything to use `activation.push` or using interior mutability so that `avm1.push` can take `&self` over `&mut self`. Did you have another solution in mind?

Reply:
> I'd like to make it harder to accidentally skip running this logic (by using avm1.push instead of stack_push).

On the contary, I'd consider minimizing complexity of push whenever it's not needed (say `action_push`). In avm2, stack push keeps popping up on profiles with non-trivial %, while on paper it should be one of the cheapest things ever.